### PR TITLE
tools/check: exit non-zero when the unit test fail

### LINKED ISSUE #32442: make gotest_in_verify_ci fail but the exit code is 0
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

See https://ci.pingcap.net/blue/organizations/jenkins/tidb-unit-test-hourly/detail/tidb-unit-test-hourly/1249/pipeline

### 2. What did you expect to see? (Required)

If any test case fail, the process exit code should not be 0

### 3. What did you see instead (Required)

Process exit 0 and the CI script success

### 4. What is your TiDB version? (Required)

master 5e52263d81bb035315b572c872df92e6fc45a719

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #32442

Problem Summary:

### What is changed and how it works?

When some test case fail, the process should exit with non-zero code.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tools/check/ut.go [hunk 1]
```diff
@@ -356,14 +356,13 @@ func main() {
 		fmt.Println("os.Getwd() error", err)
 	}
 
+	var isSucceed bool
 	if len(os.Args) == 1 {
 		// run all tests
-		cmdRun()
```
Comment (tiancaiamao) [inline tools/check/ut.go:361]: The error is ignored before this change.

**Review by hawkingrei — APPROVED** [review]

**Review by dianqihanwangzi — APPROVED** [review]
Lgtm

**Review by Defined2014 — APPROVED** [review]

File: tools/check/ut.go [hunk 2]
```diff
@@ -356,14 +356,13 @@ func main() {
...
-		return
+		isSuccess = cmdRun()
```
Comment (Defined2014) [inline tools/check/ut.go:362]:
```suggestion
		isSucceed = cmdRun()
```

**Pull request merged**